sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/ingestion/http/IngestionHttp.java

```diff
@@ -82,6 +82,12 @@
     private static final int READ_TIMEOUT = 20000;
 
     /**
+     * Regex pattern for application secret in header.
```

Comment:
I understand you you are saying and I also concerned about the format that will eventually get changed. The reason of using regex was to have simple lines of code instead of several and just implemented based on current format of app secret (having dashes looks better than all asterisks). 😄
Instead of revisiting here when the format changes, I'll simply check the length and hide the first 3/4 of the string. This way is more flexible if the app secret length is shorter. If you have different opinion and just want to hide/show fixed length, let me know. That way also works.